Finjector.Core/Services/AggieEnterpriseService.cs

```diff
@@ -0,0 +1,273 @@
+﻿using Finjector.Core.Models;
+using Microsoft.Extensions.Options;
+using AggieEnterpriseApi;
+using AggieEnterpriseApi.Extensions;
+using AggieEnterpriseApi.Validation;
+
+
+namespace Finjector.Core.Services
+{
+    public interface IAggieEnterpriseService
+    {
+        //Returns Chart type (includes invalid value)
+        FinancialChartStringType GetChartType(string segmentString);
+
+        Task<AeDetails> GetAeDetailsAsync(string segmentString);
+    }
+
+    public class AggieEnterpriseService : IAggieEnterpriseService
+    {
+        private readonly FinancialOptions _financialOptions;
+        private IAggieEnterpriseClient _apiClient;
+
+        public AggieEnterpriseService(IOptions<FinancialOptions> options)
+        {
+            _financialOptions = options.Value;
+            _apiClient = GraphQlClient.Get(_financialOptions.ApiUrl!, _financialOptions.TokenEndpoint!, _financialOptions.ConsumerKey!, _financialOptions.ConsumerSecret!, $"{_financialOptions.ScopeApp}-{_financialOptions.ScopeEnv}");
+        }
+
+        public async Task<AeDetails> GetAeDetailsAsync(string segmentString)
+        {
+            AeDetails aeDetails = new AeDetails();
+            aeDetails.ChartType = string.IsNullOrEmpty(segmentString) ? FinancialChartStringType.Invalid.ToString() : GetChartType(segmentString).ToString().ToUpper();

```

Comment:
I think we should instead just define some static strings on the `Chart` domain class, like we usually do, instead of using these enums.